src/test/java/net/sf/jabref/logic/util/version/VersionTest.java

```diff
@@ -0,0 +1,114 @@
+        assertEquals(versionText, version.getFullVersion());
+        assertEquals(1, version.getMajor());
+        assertEquals(0, version.getMinor());
+        assertEquals(0, version.getPatch());
+        assertFalse(version.isDevVersion());
+    }
+
+    @Test
+    public void versionTwoDigits() {
+        String versionText = "1.2";
+        Version version = new Version(versionText);
+        assertEquals(versionText, version.getFullVersion());
+        assertEquals(1, version.getMajor());
+        assertEquals(2, version.getMinor());
+        assertEquals(0, version.getPatch());
+        assertFalse(version.isDevVersion());
+    }
+
+    @Test
+    public void versionThreeDigits() {
+        String versionText = "1.2.3";
+        Version version = new Version(versionText);
+        assertEquals(versionText, version.getFullVersion());
+        assertEquals(1, version.getMajor());
+        assertEquals(2, version.getMinor());
+        assertEquals(3, version.getPatch());
+        assertFalse(version.isDevVersion());
+    }
+
+    @Test
+    public void versionOneDigitDevVersion() {
+        String versionText = "1dev";
+        Version version = new Version(versionText);
+        assertEquals(versionText, version.getFullVersion());
+        assertEquals(1, version.getMajor());
+        assertEquals(0, version.getMinor());
+        assertEquals(0, version.getPatch());
+        assertTrue(version.isDevVersion());
+    }
+
+    @Test
+    public void versionTwoDigitDevVersion() {
+        String versionText = "1.2dev";
+        Version version = new Version(versionText);
+        assertEquals(versionText, version.getFullVersion());
+        assertEquals(1, version.getMajor());
+        assertEquals(2, version.getMinor());
+        assertEquals(0, version.getPatch());
+        assertTrue(version.isDevVersion());
+    }
+
+    @Test
+    public void versionThreeDigitDevVersion() {
+        String versionText = "1.2.3dev";
+        Version version = new Version(versionText);
+        assertEquals(versionText, version.getFullVersion());
+        assertEquals(1, version.getMajor());
+        assertEquals(2, version.getMinor());
+        assertEquals(3, version.getPatch());
+        assertTrue(version.isDevVersion());
+    }
+
+    @Test
+    public void versionNewerThan() {
+        Version olderVersion = new Version("2.4");
+        Version newerVersion = new Version("4.2");
+        assertTrue(newerVersion.isNewerThan(olderVersion));
+    }
+
+    @Test
+    public void versionEqualVersionsWithDev() {
+        Version latestVersion = new Version("4.2");
+        Version currentVersion = new Version("4.3dev");
+        assertFalse(currentVersion.isNewerThan(latestVersion));
+    }
+
+    @Test
+    public void versionNewerThanDev() {
+        Version latestVersion = new Version("4.2.1");
+        Version currentVersion = new Version("4.3dev");
+        assertTrue(latestVersion.isNewerThan(currentVersion));
```

Comment:
Currently the latest release is 3.4, and the version in the gradle file says im working on 3.5dev.
Does the minor version gets incremented if its a dev version or exactly how do you do it?